portfolio/src/main/java/com/google/sps/servlets/FunFactsServlet.java

```diff
@@ -0,0 +1,40 @@
+package com.google.sps.servlets;
+
+import java.util.ArrayList;
+import java.io.IOException;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
+@WebServlet("/funfacts")
+public class FunFactsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> funFacts = new ArrayList<String>();
+    funFacts.add("Every year, my family and I visit Rosarito, Mexico.");
+    funFacts.add("My favorite movie-theater-snack is Trolli sour gummies.");
+    funFacts.add("To de-stress, I enjoy listening to music and painting.");
+    funFacts.add("I enjoy looking at other people's office setups.");
+    funFacts.add("I am fluent in English and Spanish.");
+    funFacts.add("My best pandemic-purchase was a Nintendo Switch.");
+    funFacts.add("My favorite genres of music are: EDM, Reggaeton, & Indie.");
+    funFacts.add("My first programming language was Java.");
+
+    String json = convertToJsonUsingGson(funFacts);
+
+    // Send the JSON as the response
```

Comment:
Nit: I think we can remove `// Send the JSON as the response`. You want comments which communicate to the reader things that are not immediately obvious when reading through the code.
As a toy example, take:
```javascript
// Print hello
console.log("hello");
```
The comment didn't communicate anything new to the reader and the reader had to parse through both lines.

On the other hand, this fun little piece of coding history trivia has comments which provide great value:
```java
 // A fast square root finding approximate algorithm. 
 float InvSqrt(float x){
        // See https://en.wikipedia.org/wiki/Fast_inverse_square_root for more details.
        float xhalf = 0.5f * x;
        int i = *(int*)&x;            // store floating-point bits in integer
        i = 0x5f3759df - (i >> 1);    // initial guess for Newton's method
        x = *(float*)&i;              // convert new bits into float
        x = x*(1.5f - xhalf*x*x);     // One round of Newton's method
        return x;
    }
```

Reply:
This makes way more sense! Thank you for your explanation.  This is something no professor has taught me and this will change the way I comment :)